templates/traefik/traefik.py

```diff
@@ -91,3 +91,9 @@ def stop(self):
         self._traefik_sal.stop()
         self.state.delete('actions', 'start')
         self.state.delete('status', 'running')
+
+    def update_endpoint(self ,etcd_endpoint):
+        self.data['etcdEndpoint'] = etcd_endpoint
+        self.state.check('actions', 'install', 'ok')
```

Comment:
his should be in a try except, we don't want to raise an error if the state is not set. Also, you should check self.state.check('actions', 'start', 'ok') we want to restart if the service is started not if it is installed. Because it might be stopped but installed and in this case we dont want to restart.
Aslo dont use self.stop and self.start so we dont change the state the service should be in, Use the sal instead. Check this as a reference https://github.com/threefoldtech/0-templates/blob/development/templates/etcd/etcd.py#L114

Reply:
I don't agree here.
This is not a monitoring method, so raising an excpetion is ok if the state is not the one required.